bika/lims/browser/dashboard/dashboard.py

```diff
@@ -582,6 +595,95 @@ def get_analyses_section(self):
                 'title': _('Analyses'),
                 'panels': out}
 
+    def get_samples_section(self):
+        out = []
+        catalog = getToolByName(self.context, 'portal_catalog')
+        query = {'portal_type': "Sample",
+                 'cancellation_state': ['active']}
+        filtering_allowed = \
+            self.context.bika_setup.getAllowDepartmentFiltering()
+        if filtering_allowed:
+            cookie_dep_uid = self.request.get('filter_by_department_info',
```

Comment:
for cookie name, better use a constant outside the class like:

```python
FILTER_BY_DEPT_COOKIE_ID = 'filter_by_department_info' 
```
Also, encapsulate this block of code like:
```python
def _get_filter_by_department_uids(self):
    if not self.context.bika_setup.getAllowedDepartmentFiltering():
        return []
    cookie_dep_uid = self.request.get(FILTER_BY_DEPT_COOKIE_ID, None)
    if not cookie_dep_uid:
        return []
    return cookie_dep_uid.split(",")

def _fill_department_uids(self, query=None):
    if not query or not isinstance(query, dict):
        return
    uids = self._get_filter_by_department_uids()
    if len(uids) == 0:
        return
    query['getDepartmentUIDs'] = {"query": uids, "operator": "or"}
```
Then, you can simply do the following:
```python
self._fill_department_uids(query)
```